shared-lib/lib-ot/tests/main.rs

```diff
@@ -1 +1,147 @@
+use lib_ot::core::{DocumentTree, NodeAttributes, NodeSubTree, Position, TransactionBuilder};
+use lib_ot::errors::OTErrorCode;
+use std::collections::HashMap;
 
+#[test]
+fn main() {
+    // Create a new arena
+    let _document = DocumentTree::new();
+}
+
+#[test]
+fn test_documents() {
+    let mut document = DocumentTree::new();
+    let transaction = {
+        let mut tb = TransactionBuilder::new(&document);
+        tb.insert_nodes_at_path(&vec![0].into(), &vec![Box::new(NodeSubTree::new("text"))]);
+        tb.finalize()
+    };
+    document.apply(transaction).unwrap();
+
+    assert!(document.node_at_path(&vec![0].into()).is_some());
+    let node = document.node_at_path(&vec![0].into()).unwrap();
+    let node_data = document.arena.get(node).unwrap().get();
+    assert_eq!(node_data.node_type, "text");
+
+    let transaction = {
+        let mut tb = TransactionBuilder::new(&document);
+        tb.update_attributes_at_path(
+            &vec![0].into(),
+            HashMap::from([("subtype".into(), Some("bullet-list".into()))]),
+        );
+        tb.finalize()
+    };
+    document.apply(transaction).unwrap();
+
+    let transaction = {
+        let mut tb = TransactionBuilder::new(&document);
+        tb.delete_node_at_path(&vec![0].into());
+        tb.finalize()
+    };
+    document.apply(transaction).unwrap();
+    assert!(document.node_at_path(&vec![0].into()).is_none());
+}
+
+#[test]
+fn test_inserts_nodes() {
```

Comment:
Try using the script-style test.  For example:
```rust
  let scripts = vec![
        MoveRow {
            from_group_index: 0,
            from_row_index: 0,
            to_group_index: 1,
            to_row_index: 1,
        },
        AssertGroupRowCount {
            group_index: 0,
            row_count: 1,
        },
        AssertGroupRowCount {
            group_index: 1,
            row_count: 3,
        },
        AssertRow {
            group_index: 1,
            row_index: 1,
            row: group.rows.get(0).unwrap().clone(),
        },
    ];
    test.run_scripts(scripts).await;
```

You could check out the enum `GroupScript` for more information.